nextcloudappstore/core/views.py

```diff
@@ -18,19 +20,48 @@ def get_context_data(self, **kwargs):
 class CategoryAppListView(ListView):
     model = App
     template_name = 'app/list.html'
+    allow_empty = True
 
     def get_queryset(self):
         category_id = self.kwargs['id']
         queryset = super().get_queryset()
+
         if category_id:
-            return queryset.filter(categories__id=category_id)
-        else:
-            return queryset
+            queryset = queryset.filter(categories__id=category_id)
+
+        if self.search_terms:
+            query = None
+
+            for term in self.search_terms:
```

Comment:
As for the translations: maybe this is a good point in time to extract the search logic into a separate function and create some tests for it.

Check out https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/core/tests/test_versioning.py#L6 and https://github.com/nextcloud/appstore/blob/master/nextcloudappstore/core/tests/__init__.py

You can create some apps first and set the appropriate language on them. Then run a test search and check if the correct number of apps is being returned :)